javaslang/src/main/java/javaslang/collection/Seq.java

```diff
@@ -585,7 +584,11 @@ default int prefixLength(Predicate<? super T> predicate) {
      * @return a Seq containing all elements of this but none of the given elements.
      * @throws NullPointerException if {@code elements} is null
      */
-    Seq<T> removeAll(Iterable<? extends T> elements);
+    default Seq<T> removeAll(Iterable<? extends T> elements) {
+        Objects.requireNonNull(elements, "elements is null");
+        final Set<T> removed = HashSet.ofAll(elements);
+        return filter(e -> !removed.contains(e));
```

Comment:
This is a great idea! However, we decide not to have delegating default implementations with return type casts in our interfaces. Instead let's add the method(s) to our `javaslang.collection.Collections` class:

```java
@SuppressWarnings("unchecked")
static <C extends Traversable<T>, T> C removeAll(C collection, Iterable<? extends T> elements) {
    Objects.requireNonNull(elements, "elements is null");
    final Set<T> removed = HashSet.ofAll(elements);
    return (C) collection.filter(e -> !removed.contains(e));
}
```

(We don't need to check if the given `collection` is null because it is for internal use only and the calling collections will always provide `this`.)

Bonus points: We isolate the unsafe cast.